### DISCUSSION TITLE
Arbitrary length arrays in command parameters

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1246] [Q&A]
Good morning,

I need to be able to send a command to F' with a parameter of arbitrary length based on the previous parameter:
```
<args>
  <arg name="n" type="U32" />
  <arg name="my_buffer" type="U8" size="n" pass_by="reference" />
</args>
``` 
Is this legal? What is the appropriate way to do this?

**Comment by timcanham** [comment]
`my_buffer`'s length is set by the `size` attribute, and you can't tie that to another parameter like `n`. There is no reason that in the implementation of your component command handler your code can't use `n` to decide how much of `my_buffer` to use.

**Reply by JackNWhite** [reply]
Thanks, @timcanham. Can I use the name of preprocessor constant defined elsewhere for `size`?
